hazelcast/include/hazelcast/client/protocol/ClientMessage.h

```diff
@@ -917,6 +917,44 @@ class HAZELCAST_API ClientMessage
         return h;
     }
 
+    template<typename T>
+    typename std::
+      enable_if<std::is_same<T, serialization::pimpl::schema>::value, T>::type
+      get()
+    {
+        using namespace serialization;
+        using namespace serialization::pimpl;
+
+        // skip begin frame
+        skip_frame();
+
+        auto type_name = get<std::string>();
+
+        std::unordered_map<std::string, field_descriptor> fields;
+        {
```

Comment:
Better to do like this:
`auto fields_list = get<std::vector<field_descriptor>>();`

and have schema constructor works with list as well.

Reply:
Agree but the problem is `field_descriptor` in Java and C++ is different, so `get<std::vector<field_descriptor>>()` is not usable.

Reply:
ok